src/core/src/core_logic/PatchInstaller.py

```diff
@@ -72,22 +76,38 @@ def start_installation(self, simulate=False):
             installed_update_count += new_installed_update_count
+            number_of_rounds = number_of_rounds + 1
 
             if package_manager.get_package_manager_setting(Constants.PACKAGE_MGR_SETTING_REPEAT_PATCH_OPERATION, False):  # We should not see this again
                 error_msg = "Unexpected repeated package manager update occurred. Please re-run the update deployment."
                 self.status_handler.add_error_to_status(error_msg, Constants.PatchOperationErrorCodes.PACKAGE_MANAGER_FAILURE)
+                perc_maintenance_window_used = maintenance_window.get_percentage_maintenance_window_used()
+                patch_installation_perf_log  = {Constants.LogStrings.TASK: Constants.LogStrings.PATCH_INSTALLATION, Constants.LogStrings.PACKAGE_MANAGER: self.package_manager_name,
+                                                Constants.LogStrings.UPDATE_RUN_SUCCESSFUL: str(update_run_successful), Constants.LogStrings.INSTALLED_UPDATE_COUNT: str(installed_update_count),
+                                                Constants.LogStrings.NUMBER_OF_ROUNDS: str(number_of_rounds), Constants.LogStrings.MAINTENANCE_WINDOW: str(maintenance_window.duration),
+                                                Constants.LogStrings.PERC_MAINTENANCE_WINDOW_USED: str(perc_maintenance_window_used), Constants.LogStrings.MAINTENANCE_WINDOW_EXCEEDED: str(maintenance_window_exceeded),
+                                                Constants.LogStrings.TASK_STATUS: Constants.LogStrings.FAILED, Constants.LogStrings.ERROR_MSG: error_msg}
+                stopwatch.stop_and_write_telemetry(str(patch_installation_perf_log))
                 raise Exception(error_msg, "[{0}]".format(Constants.ERROR_ADDED_TO_STATUS))
 
         self.composite_logger.log("\nInstalled update count: " + str(installed_update_count) + " (including dependencies)")
 
+        perc_maintenance_window_used = maintenance_window.get_percentage_maintenance_window_used()
+        patch_installation_perf_log  = {Constants.LogStrings.TASK: Constants.LogStrings.PATCH_INSTALLATION, Constants.LogStrings.PACKAGE_MANAGER: self.package_manager_name,
+                                        Constants.LogStrings.UPDATE_RUN_SUCCESSFUL: str(update_run_successful), Constants.LogStrings.INSTALLED_UPDATE_COUNT: str(installed_update_count),
+                                        Constants.LogStrings.NUMBER_OF_ROUNDS: str(number_of_rounds), Constants.LogStrings.MAINTENANCE_WINDOW: str(maintenance_window.duration),
+                                        Constants.LogStrings.PERC_MAINTENANCE_WINDOW_USED: str(perc_maintenance_window_used), Constants.LogStrings.MAINTENANCE_WINDOW_EXCEEDED: str(maintenance_window_exceeded),
+                                        Constants.LogStrings.TASK_STATUS: Constants.LogStrings.SUCCEEDED, Constants.LogStrings.ERROR_MSG: ""}
```

Comment:
Similar to the comment in PatchAssessor, this can be extracted out to a function set_installer_perf_logs(), called on line 101 and 91 with the relevant values.